lib/OpenQA/WebAPI/Controller/API/V1/JobTemplate.pm

```diff
@@ -247,51 +247,54 @@ sub update ($self) {
         $user_errors
           = $self->app->validate_yaml($data, $validation->param('schema'), $self->app->log->level eq 'debug');
     }
-    catch {
+    catch ($e) {
         # Push the exception to the list of errors without the trailing new line
-        push @$user_errors, substr($_, 0, -1);
-    };
+        push @$user_errors, substr($e, 0, -1);
+    }
     return $self->respond_to(json => {json => {error => $user_errors}, status => 400}) if @$user_errors;
 
     my $schema = $self->schema;
     my $job_groups = $schema->resultset('JobGroups');
     my $json = {};
     my @server_errors;
     try {
-        my $id = $self->param('id');
-        my $name = $validation->param('name');
-        my $job_group = $job_groups->find($id ? {id => $id} : ($name ? {name => $name} : undef),
-            {select => [qw(id name template)]});
-        return push @$user_errors, 'Job group ' . ($name // $id) . ' not found' unless $job_group;
-        my $group_id = $job_group->id;
-        $json->{job_group_id} = $group_id;
-        # Backwards compatibility: ID used to mean group ID on this route
-        $json->{id} = $group_id;
-
-        if (my $reference = $validation->param('reference')) {
-            my $template = $job_group->to_yaml;
-            $json->{template} = $template;
-            # Compare with no regard for trailing whitespace
-            chomp $template;
-            chomp $reference;
-            return push @$user_errors, 'Template was modified' unless $template eq $reference;
-        }
+        {    # extra block for early returns via "last"
+            my $id = $self->param('id');
+            my $name = $validation->param('name');
+            my $job_group = $job_groups->find($id ? {id => $id} : ($name ? {name => $name} : undef),
+                {select => [qw(id name template)]});
+            push @$user_errors, 'Job group ' . ($name // $id) . ' not found' && last unless $job_group;
+            my $group_id = $job_group->id;
+            $json->{job_group_id} = $group_id;
+            # Backwards compatibility: ID used to mean group ID on this route
+            $json->{id} = $group_id;
+
+            if (my $reference = $validation->param('reference')) {
+                my $template = $job_group->to_yaml;
+                $json->{template} = $template;
+                # Compare with no regard for trailing whitespace
+                chomp $template;
+                chomp $reference;
+                push @$user_errors, 'Template was modified' and last unless $template eq $reference;
```

Comment:
so this is still four levels of indentation which I consider more than desired. I am ok with the change but would still be interested if we can do better. I might try as a "evening fun project" so don't wait for me :)